tokio-trace/src/span.rs

```diff
@@ -124,6 +124,10 @@ pub struct Data {
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
 pub struct Id(u64);
 
+pub trait AsId {
+    fn as_id(&self) -> Option<Id>;
```

Comment:
Abstracts over anything that may have a span ID, so we can add a follows-from annotation from a `Span` handle, an `Id`, and so on. I was hoping there was a trait in the standard library that could be used here, but this seemed simpler.

Reply:
Yeah, I guess this depends on https://github.com/rust-lang/rust/issues/33417 landing.